torch/csrc/autograd/functions/batch_normalization.cpp

```diff
@@ -73,14 +74,14 @@ auto BatchNormForward::apply(const variable_list& inputs) -> variable_list {
         state,
         torch::cudnn::getCudnnHandle(),
         torch::cudnn::getCudnnDataType(input),
-        (THVoidTensor*)input.unsafeGetTH(false),
-        (THVoidTensor*)output.unsafeGetTH(false),
-        (THVoidTensor*)weight.unsafeGetTH(false),
-        (THVoidTensor*)bias.unsafeGetTH(false),
-        (THVoidTensor*)running_mean.unsafeGetTH(false),
-        (THVoidTensor*)running_var.unsafeGetTH(false),
-        (THVoidTensor*)save_mean.unsafeGetTH(false),
-        (THVoidTensor*)save_std.unsafeGetTH(false),
+        input_data,
+        output,
+        weight.opt_data(),
+        bias.opt_data(),
```

Comment:
Can we please use `_var` suffixes for `at::Tensors` that are Variables? It's really hard to figure out what are the types of these things

Reply:
And to make matters worse, the implicit conversion from Variable to Tensor means you won't even notice if you get it wrong ;) (yes I'll do it)